cardano-cli/src/Cardano/CLI/Compatible/Run.hs

```diff
@@ -32,6 +34,19 @@ runAnyCompatibleCommand (AnyCompatibleCommand cmd) = runCompatibleCommand cmd
 
 runCompatibleCommand :: CompatibleCommand era -> ExceptT CompatibleCmdError IO ()
 runCompatibleCommand (CompatibleTransactionCmd txCmd) =
-  firstExceptT CompatibleTransactionError $ runCompatibleTransactionCmd txCmd
+  liftIO $
+    executeRio
+      (runCompatibleTransactionCmd txCmd)
 runCompatibleCommand (CompatibleGovernanceCmds govCmd) =
   firstExceptT CompatibleGovernanceError $ runCompatibleGovernanceCmds govCmd
+
+-- NB: We should not be handling exceptions here. However this is temporary
+-- as we migrate away from ExceptT.
+executeRio :: RIO () () -> IO ()
+executeRio r = do
+  runRIO () r
+    `catch` ( \e ->
+                case fromException e of
+                  Just custom@(CustomCliException{}) -> putStrLn $ displayException custom
```

Comment:
Dead code. `CustomException` is never thrown.

Reply:
This is meant to be a demonstration for the ADR. I am aware `CustomCliException` is not thrown. I am showing how we can in theory use `CustomCliException` in our code to render the error and provide a callstack.